libexfat/io.c

```diff
@@ -20,29 +20,40 @@
 	51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
+#define _GNU_SOURCE /* for fcntl.h fallocate */
```

Comment:
It would be cleaner to detect fallocate() with autotools.

Reply:
`fcntl.h` only declares the non-standard `fallocate` if `_GNU_SOURCE` is defined. I added it here since it does have to come before all the headers and it doesn't hurt on other libcs.

Still, it's cleaner to switch to `AC_GNU_SOURCE` (which will always define `_GNU_SOURCE` if available) and detecting with `AC_CHECK_DECLS([fallocate], [], [], [#include <fcntl.h>])`.